test/clj/rems/test_event_notification.clj

```diff
@@ -0,0 +1,169 @@
+    (with-redefs [rems.config/env (assoc rems.config/env
+                                         :event-notification-targets [{:url (str (:uri server) "/created")
+                                                                       :event-types [:application.event/created]}
+                                                                      {:url (str (:uri server) "/all")}])]
+      (let [get-notifications #(doall
+                                (for [r (stub/recorded-requests server)]
+                                  {:path (:path r)
+                                   :data (-> r
+                                             :body
+                                             (get "content")
+                                             json/parse-string
+                                             (dissoc-in [:event/application :application/events])
+                                             ;; catalogue-item/start is set by the db and can't be easily fixed
+                                             (dissoc-in [:event/application :application/resources 0 :catalogue-item/start]))}))
+            form-id (test-data/create-form! {:form/title "notifications"
+                                             :form/fields [{:field/type :text
+                                                            :field/id "field-1"
+                                                            :field/title {:en "text field"}
+                                                            :field/optional false}]})
+            handler "handler"
+            workflow-id (test-data/create-workflow! {:title "wf"
+                                                     :handlers [handler]
+                                                     :type :workflow/default})
+            ext-id "resres"
+            res-id (test-data/create-resource! {:resource-ext-id ext-id})
+            cat-id (test-data/create-catalogue-item! {:form-id form-id
+                                                      :resource-id res-id
+                                                      :workflow-id workflow-id})
+            applicant "alice"
+            app-id (:application-id (command/command! {:type :application.command/create
+                                                       :actor applicant
+                                                       :time (time/date-time 2001)
+                                                       :catalogue-item-ids [cat-id]}))]
+        (testing "no notifications before outbox is processed"
+          (is (= []
+                 (stub/recorded-requests server))))
+        (event-notification/process-outbox!)
+        (testing "created event gets sent to both endpoints"
+          (let [notifications (get-notifications)]
+            (is (= 2 (count notifications)))
+            (is (= #{"/created" "/all"}
+                   (set (map :path notifications))))
+            (is (= {:application/external-id "2001/1"
```

Comment:
Would it make sense to compare this to the get raw output like it is specified in documentation? Seems like overkill to test everything about the application here. Especially since it's not a very rich application. Maybe a gold standard test in the raw API and compare to that here?

Reply:
you mean making a var for the gold standard and using it in two places? I think event notifications are important enough to warrant a gold standard test so that's why I added this.

Reply:
I think the situation in this PR is that now there is
- a new raw API
- an outbound event that produces mostly identical output to raw API response
- gold standard outbound event detail test

I think we are missing
- explicit testing of raw response matches outbound event as promised by doc
- explicit testing of raw API response details, gold standard for it?

I propose we 
- test the raw API response exactly in a gold standard test and
- not test the outbound event like this, but so that raw API part is compared to raw API response for the exact same situation creating an explicit test for the documented behavior
- bonus: add slightly more detail to the raw API gold standard test

Reply:
yeah that's a good idea. done